traits/src/asset_registry.rs

```diff
@@ -0,0 +1,14 @@
+use frame_support::pallet_prelude::*;
+use xcm::latest::prelude::*;
+
+pub trait WeightToFeeConverter {
+	fn convert_weight_to_fee(location: &MultiLocation, weight: Weight) -> Option<u128>;
+}
+
+pub trait FixedConversionRateProvider {
+	fn get_fee_per_second(location: &MultiLocation) -> Option<u128>;
+}
+
+pub trait AssetProcessor<AssetId, Metadata> {
+	fn process_asset(id: Option<AssetId>, asset_metadata: Metadata) -> Result<(AssetId, Metadata), DispatchError>;
```

Comment:
rename this method to `pre_register(..)` and add another method named: `post_register()`
```rust
fn pre_register(id: Option<AssetId>, asset_metadata: Metadata) -> Result<(AssetId, Metadata), DispatchError>;
fn post_register(id: Option<AssetId>, asset_metadata: Metadata) -> Result<(AssetId, Metadata), DispatchError>;
```

As customized parachains may want to do additional behaviour after asset registed.

Reply:
Done (except the type is `fn post_register(_id: AssetId, _asset_metadata: Metadata) -> Result<(), DispatchError>`).

Currently they are only called in `register_asset`, not in `update_asset`. Not sure if we should have hooks there as well..

Reply:
`update_asset` don't need hook for now. but if someone have those customize, it can be add then.